### PR TITLE
Enable GitHub's dependabot for automatic dependency updates

### PR DESCRIPTION
**Pull request by shonfeder**
Closes: N/A

## Description

Dependabot scans dependencies daily for updates and will open PRs
to upgrade the deps (which will then run through CI), if newer
versions are published.

For details, see
https://docs.github.com/en/github/administering-a-repository/about-github-dependabot-version-updates

______

For contributor use:

- [ ] Unit tests written + N/A
- [ ] Added test to CI if applicable + N/A 
- [ ] Updated CHANGELOG_PENDING.md + N/A
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work. + N/A
- [X] Updated relevant documentation (`docs/`) and code comments
- [X] Re-reviewed `Files changed` in the Github PR explorer + N/A

### CODE REVIEW

**Comment by shonfeder** [comment]
This is just an off-the-cuff suggestion, feel free to close/reject if not desired for this repo :)

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+version: 2
+updates:
+  # Enable version updates for npm
+  - package-ecosystem: "cargo"
+    # Look for Cargo `.toml` and `.lock` files in the `root` directory
+    directory: "/"
+    # Check the cargo registry for updates every day (weekdays)
+    schedule:
+      interval: "daily"
```
Comment (adizere) [inline .github/dependabot.yml:12]: Maybe we can do it weekly? The repo is still going through breaking changes all the time, and we have a chronic issue with too many outstanding PRs, so would be nice to minimize the amount of work we get from the bot.

**Review by adizere — APPROVED** [review]
Any experience with the dependabot @romac or @ancazamfir ? A priori seems like a good idea :)

**Comment by ancazamfir** [comment]
Good idea, I don't have any experience though. Was trying to find how tendermint-rs has enabled this, can anyone point me to their approach?

**Comment by shonfeder** [comment]
afaik, tendermint-rs does not have this configured. To enable it -- if we want to -- we just merge this config file into the trunk.

File: .github/dependabot.yml (hunk 1 above)
Comment (shonfeder) [reply .github/dependabot.yml:12]: Sure, however you'd prefer! One thing to note: configuring it to only check weekly won't reduce the amount of work required, it'll just mean all the work shows up at once on the same day ;) 

Regarding the workload, it should be minimal in any case. Usually you just either merge in the PRs if CI passes, or postpone the update (via a special comment on the PR or just closing the PR) and create an issue to fix the compatibility with the library.

However, if updated or dealing with outdated dependencies isn't something that's an issue, then maybe you don't need this enabled here! :)

**Pull request merged**